probe-rs/src/debug/mod.rs

```diff
@@ -342,41 +368,6 @@ impl<'debuginfo, 'probe, 'core> Iterator for StackFrameIterator<'debuginfo, 'pro
             }
         }
 
```

Comment:
Can you please restore and update some of the comments that you removed here? This unwind function has multiple steps which makes future maintenance very tricky if the logic and sequence for each is not clearly explained.

Reply:
Is there anything specific that is unclear? I'll add some comments, but clarifying concrete questions is probably most helpful.

Reply:
Sorry if my request wasn't clear.  I noticed that you deleted the comments that used to be marked "Part 2:", and "Part 2-a:"   ... that leaves the rest of the narrative to start at "Part 2-b:". It would be nice if you could insert something to describe the transition from "Part 1" to "Part 2-b", or at the least restore the one that was marked "Part 2:" (it was kind of a heading), and rename the rest of the comments to be alphabetically correct.  It is not critical, but it took me hours to understand the original unwind logic because the code is not really self explanatory, so I'd like to at least provide nominal breadcrumbs for the next person who has to modify this code :)

Reply:
Thanks, that clarifies it. I will add the heading back. 

In the long run, I think we should split up the function into multiple functions, for me that's clearer than a single very long function. But that's probably better done later.

Reply:
Agreed. I think the right time to do that is probably when we make it fully workable for RISCV.